cosmetic changes

komatits opened the pull request.

buildbot-princeton commented:
The files changed require a buildbot test. Buildbot has started.

buildbot-princeton commented:
Your changes failed the buildbot test. See [build 528](http://specfem2d-buildbot.princeton.edu/builders/specfem2d_builder/builds/528). Fix your mistakes and push your changes back to your branch komatits/specfem2d:devel. This pull-request will be reopened automatically.         You don't need to create a new pull request

luet commented:
I'll work on it this morning.

komatits commented:
Thanks!

On 04/24/2017 03:26 PM, David Luet wrote:
> I'll work on it this morning.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-296667136>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFjDKT4x5K91SRB6L6Gs3_3CqaJwyEWLks5rzKMGgaJpZM4NGITG>.
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

komatits commented:
accepting manually then.

On 04/24/2017 03:11 PM, buildbot-princeton wrote:
> Your changes failed the buildbot test. See build 528
> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_builder/builds/528>.
> Fix your mistakes and push your changes back to your branch
> komatits/specfem2d:devel. This pull-request will be reopened
> automatically. You don't need to create a new pull request
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-296662797>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFjDKZsQJWz7Pv-0zCGehkH2BPDHy2lNks5rzJ-dgaJpZM4NGITG>.
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

luet commented:
We are running into a problem with the correlation test. Some of the vectors are just filled with zeros:

```
import numpy as np
a=np.array([0,0,0])
b=np.array([0, 0, 0])
corr_mat=np.corrcoef(a, b)
corr=np.min(corr_mat)
```
returns `corr_mat`
```
array([[ nan,  nan],
       [ nan,  nan]])
```
and if you look at the seismograms: `AA.S0022.BXX.semd` and `AA.S0022.BXZ.semd` the displacements are all 0.

[numpy.corrcoef](https://docs.scipy.org/doc/numpy/reference/generated/numpy.corrcoef.html) uses the [Pearson correlation coefficient](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient), which involves a division by 0 when the vectors are all zeros.

This happens line 58 of [CheckExamples.py](https://github.com/luet/CompareSeismo/blob/master/CheckExamples.py#L58).

I am not clear on how we need to handle this problem. We could just ignore the NANs but I think we might miss some important cases.

I am also not clear on how good the correlation test is:
```
a=np.array([0., 1.,0.])
b=np.array([0., .1, 0.])
corr_mat=np.corrcoef(a, b)
```
returns
```
array([[ 1.,  1.],
       [ 1.,  1.]])
```
which is not what we want.

komatits commented:
Hi David,

OK thanks, I get the point; it is a pressure receiver that is located on 
an acoustic free surface and thus pressure is identically zero there. 
That receiver was added last week.

I guess what you can do is test the seismograms to see if they are 
identically zero (max(abs(seismogram)) < 1.e-15 or similar) and if so 
exclude them from the list to be tested
(then check that both old and new are identically zero instead).

We cannot simply remove receiver #22 from the test, because such 
receivers located on an edge of the mesh can (and do) happen.

Thanks,
Best wishes,

Dimitri.

On 04/24/2017 05:13 PM, David Luet wrote:
> We are running into a problem with the correlation test. Some of the
> vectors are just filled with zeros:
>
> |import numpy as np a=np.array([0,0,0]) b=np.array([0, 0, 0])
> corr_mat=np.corrcoef(a, b) corr=np.min(corr_mat) |
>
> returns |corr_mat|
>
> |array([[ nan, nan], [ nan, nan]]) |
>
> and if you look at the seismograms: |AA.S0022.BXX.semd| and
> |AA.S0022.BXZ.semd| the displacements are all 0.
>
> numpy.corrcoef
> <https://docs.scipy.org/doc/numpy/reference/generated/numpy.corrcoef.html>
> uses the Pearson correlation coefficient
> <https://en.wikipedia.org/wiki/Pearson_correlation_coefficient>, which
> involves a division by 0 when the vectors are all zeros.
>
> This happens line 58 of CheckExamples.py
> <https://github.com/luet/CompareSeismo/blob/master/CheckExamples.py#L58>.
>
> I am not clear on how we need to handle this problem. We could just
> ignore the NANs but I think we might miss some important cases.
>
> I am also not clear on how good the correlation test is:
>
> |a=np.array([0., 1.,0.]) b=np.array([0., .1, 0.]) corr_mat=np.corrcoef(a,
> b) |
>
> returns
>
> |array([[ 1., 1.], [ 1., 1.]]) |
>
> which is not what we want.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-296700670>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFjDKdKJxAMTGylfYnHQzbgW4ZWTKeTOks5rzLwDgaJpZM4NGITG>.
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

luet commented:
I did what you described, along with using the new seismograms as reference. That solved the problem with `simple_topography_and_also_a_simple_fluid_layer`.

But then I run into another issue with `axisymmetric_case_AXISYM_option`. See [build 261](http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/261/steps/axisymmetric_case_AXISYM_option/logs/stdio) and the [PDFs](http://seismobot.princeton.edu/nightly_build_pdf/specfem2d/axisymmetric_case_AXISYM_option/axisymmetric_case_AXISYM_option.pdf).
Should I just use the new seismograms as reference?
Thanks,
David

komatits commented:
Hi David,

Regarding the AXISYM option let me cc Alexis, who is the main developer.

The logs you sent say:

awk: cmd. line:1: fatal: cannot open file `Par_file_axisym' for reading 
(No such file or directory)

which is weird because the parameter file of that example is called 
Par_file, not Par_file_axisym. Are you using an old version of the repo?

The logs also contain this:

Open MPI failed to open the /dev/knem device due to a local error.
Please check with your system administrator to get the problem fixed,
or set the btl_sm_use_knem MCA parameter to 0 to run without /dev/knem
support.

thus it seems MPI has a problem on the machine that BuildBot uses.

Thanks,
Dimitri.

On 04/24/2017 08:46 PM, David Luet wrote:
> I did what you described, along with using the new seismograms as
> reference. That solved the problem with
> |simple_topography_and_also_a_simple_fluid_layer|.
>
> But then I run into another issue with
> |axisymmetric_case_AXISYM_option|. See build 261
> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/261/steps/axisymmetric_case_AXISYM_option/logs/stdio>
> and the PDFs
> <http://seismobot.princeton.edu/nightly_build_pdf/specfem2d/axisymmetric_case_AXISYM_option/axisymmetric_case_AXISYM_option.pdf>.
> Should I just use the new seismograms as reference?
> Thanks,
> David
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-296786701>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFjDKaTpwCp2lkrYf3NRv3mCH38Xk2POks5rzO3qgaJpZM4NGITG>.
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

luet commented:
The problem with the `awk` command failing is not a problem. I was trying to replace the default `nproc=16` by `nproc=8`. There a only 16 cores on the machine and I wanted to leave 8 for other tasks.

We have already talked about the MPI message many times. It's a warning not an error and it only affects performance, not the results. If you look at [build 256](http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/256/steps/axisymmetric_case_AXISYM_option/logs/stdio), the last successful test, you'll see the same warning about MPI.

The last successful commit was 2e7d50c and all the commits after that were done by @komatits. Could it be that one of the default values you changed affect this example?

komatits commented:
Hi,

Yes, we changed PML_parameter_adjustment to .true. when NSOURCES == 1, 
following a suggestion by Zhinan; This is supposed to improve the PMLs 
and thus the seismograms (and thus change them, and thus the correlation 
test will fail).

However for AXISYM it is not clear if the new seismograms are better or 
worse. Alexis could you please tell us? (the PDFs are in the link below, 
or else redo your AXISYM test with and without this at line 494 of 
src/specfem2D/pml_init.F90 :

   if (NSOURCES == 1) then
     PML_parameter_adjustment = .true.
   else
     PML_parameter_adjustment = .false.
   endif


Thanks,
Dimitri.


On 04/25/2017 05:43 PM, David Luet wrote:
> The problem with the |awk| command failing is not a problem. I was
> trying to replace the default |nproc=16| by |nproc=8|. There a only 16
> cores on the machine and I wanted to leave 8 for other tasks.
>
> We have already talked about the MPI message many times. It's a warning
> not an error and it only affects performance, not the results. If you
> look at build 256
> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/256/steps/axisymmetric_case_AXISYM_option/logs/stdio>,
> the last successful test, you'll see the same warning about MPI.
>
> The last successful commit was 2e7d50c
> <https://github.com/geodynamics/specfem2d/commit/2e7d50c4ee1ca5a4172ecd70196d199429ccfe8a>
> and all the commits after that were done by @komatits
> <https://github.com/komatits>. Could it be that one of the default
> values you changed affect this example?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-297073097>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFjDKZ-kTEQYf77igEO5qVv7a7FnWUx1ks5rzhS-gaJpZM4NGITG>.
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

komatits commented:
Hi all,

Easy to revert if needed. Alexis just let me know if we should keep it 
or revert it.
Let me also cc Zhinan.

cheers
Dimitri

On 04/25/2017 07:11 PM, Dimitri Komatitsch wrote:
>
> Hi,
>
> Yes, we changed PML_parameter_adjustment to .true. when NSOURCES == 1,
> following a suggestion by Zhinan; This is supposed to improve the PMLs
> and thus the seismograms (and thus change them, and thus the correlation
> test will fail).
>
> However for AXISYM it is not clear if the new seismograms are better or
> worse. Alexis could you please tell us? (the PDFs are in the link below,
> or else redo your AXISYM test with and without this at line 494 of
> src/specfem2D/pml_init.F90 :
>
>   if (NSOURCES == 1) then
>     PML_parameter_adjustment = .true.
>   else
>     PML_parameter_adjustment = .false.
>   endif
>
>
> Thanks,
> Dimitri.
>
>
> On 04/25/2017 05:43 PM, David Luet wrote:
>> The problem with the |awk| command failing is not a problem. I was
>> trying to replace the default |nproc=16| by |nproc=8|. There a only 16
>> cores on the machine and I wanted to leave 8 for other tasks.
>>
>> We have already talked about the MPI message many times. It's a warning
>> not an error and it only affects performance, not the results. If you
>> look at build 256
>> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/256/steps/axisymmetric_case_AXISYM_option/logs/stdio>,
>>
>> the last successful test, you'll see the same warning about MPI.
>>
>> The last successful commit was 2e7d50c
>> <https://github.com/geodynamics/specfem2d/commit/2e7d50c4ee1ca5a4172ecd70196d199429ccfe8a>
>>
>> and all the commits after that were done by @komatits
>> <https://github.com/komatits>. Could it be that one of the default
>> values you changed affect this example?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-297073097>,
>>
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AFjDKZ-kTEQYf77igEO5qVv7a7FnWUx1ks5rzhS-gaJpZM4NGITG>.
>>
>>
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

komatits commented:
Hi all,

Alexis has found that the problem is not related to 
PML_parameter_adjustment, it is related to a conflict in a commit I made 
to merge viscoacoustic routines by Quentin with slightly older database 
format changes from Etienne. It seems something went slightly wrong in 
the merge (which was uneasy to do). I will debug that tomorrow and will 
keep you posted.

Cheers,
Dimitri/.

On 04/25/2017 07:20 PM, Dimitri Komatitsch wrote:
>
> Hi all,
>
> Easy to revert if needed. Alexis just let me know if we should keep it
> or revert it.
> Let me also cc Zhinan.
>
> cheers
> Dimitri
>
> On 04/25/2017 07:11 PM, Dimitri Komatitsch wrote:
>>
>> Hi,
>>
>> Yes, we changed PML_parameter_adjustment to .true. when NSOURCES == 1,
>> following a suggestion by Zhinan; This is supposed to improve the PMLs
>> and thus the seismograms (and thus change them, and thus the correlation
>> test will fail).
>>
>> However for AXISYM it is not clear if the new seismograms are better or
>> worse. Alexis could you please tell us? (the PDFs are in the link below,
>> or else redo your AXISYM test with and without this at line 494 of
>> src/specfem2D/pml_init.F90 :
>>
>>   if (NSOURCES == 1) then
>>     PML_parameter_adjustment = .true.
>>   else
>>     PML_parameter_adjustment = .false.
>>   endif
>>
>>
>> Thanks,
>> Dimitri.
>>
>>
>> On 04/25/2017 05:43 PM, David Luet wrote:
>>> The problem with the |awk| command failing is not a problem. I was
>>> trying to replace the default |nproc=16| by |nproc=8|. There a only 16
>>> cores on the machine and I wanted to leave 8 for other tasks.
>>>
>>> We have already talked about the MPI message many times. It's a warning
>>> not an error and it only affects performance, not the results. If you
>>> look at build 256
>>> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/256/steps/axisymmetric_case_AXISYM_option/logs/stdio>,
>>>
>>>
>>> the last successful test, you'll see the same warning about MPI.
>>>
>>> The last successful commit was 2e7d50c
>>> <https://github.com/geodynamics/specfem2d/commit/2e7d50c4ee1ca5a4172ecd70196d199429ccfe8a>
>>>
>>>
>>> and all the commits after that were done by @komatits
>>> <https://github.com/komatits>. Could it be that one of the default
>>> values you changed affect this example?
>>>
>>> —
>>> You are receiving this because you were mentioned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-297073097>,
>>>
>>>
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe-auth/AFjDKZ-kTEQYf77igEO5qVv7a7FnWUx1ks5rzhS-gaJpZM4NGITG>.
>>>
>>>
>>>
>>
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr

komatits commented:
Fixed.

Dimitri.

On 04/27/2017 03:59 PM, Dimitri Komatitsch wrote:
>
> Hi all,
>
> Alexis has found that the problem is not related to
> PML_parameter_adjustment, it is related to a conflict in a commit I made
> to merge viscoacoustic routines by Quentin with slightly older database
> format changes from Etienne. It seems something went slightly wrong in
> the merge (which was uneasy to do). I will debug that tomorrow and will
> keep you posted.
>
> Cheers,
> Dimitri/.
>
> On 04/25/2017 07:20 PM, Dimitri Komatitsch wrote:
>>
>> Hi all,
>>
>> Easy to revert if needed. Alexis just let me know if we should keep it
>> or revert it.
>> Let me also cc Zhinan.
>>
>> cheers
>> Dimitri
>>
>> On 04/25/2017 07:11 PM, Dimitri Komatitsch wrote:
>>>
>>> Hi,
>>>
>>> Yes, we changed PML_parameter_adjustment to .true. when NSOURCES == 1,
>>> following a suggestion by Zhinan; This is supposed to improve the PMLs
>>> and thus the seismograms (and thus change them, and thus the correlation
>>> test will fail).
>>>
>>> However for AXISYM it is not clear if the new seismograms are better or
>>> worse. Alexis could you please tell us? (the PDFs are in the link below,
>>> or else redo your AXISYM test with and without this at line 494 of
>>> src/specfem2D/pml_init.F90 :
>>>
>>>   if (NSOURCES == 1) then
>>>     PML_parameter_adjustment = .true.
>>>   else
>>>     PML_parameter_adjustment = .false.
>>>   endif
>>>
>>>
>>> Thanks,
>>> Dimitri.
>>>
>>>
>>> On 04/25/2017 05:43 PM, David Luet wrote:
>>>> The problem with the |awk| command failing is not a problem. I was
>>>> trying to replace the default |nproc=16| by |nproc=8|. There a only 16
>>>> cores on the machine and I wanted to leave 8 for other tasks.
>>>>
>>>> We have already talked about the MPI message many times. It's a warning
>>>> not an error and it only affects performance, not the results. If you
>>>> look at build 256
>>>> <http://specfem2d-buildbot.princeton.edu/builders/specfem2d_nightly/builds/256/steps/axisymmetric_case_AXISYM_option/logs/stdio>,
>>>>
>>>>
>>>>
>>>> the last successful test, you'll see the same warning about MPI.
>>>>
>>>> The last successful commit was 2e7d50c
>>>> <https://github.com/geodynamics/specfem2d/commit/2e7d50c4ee1ca5a4172ecd70196d199429ccfe8a>
>>>>
>>>>
>>>>
>>>> and all the commits after that were done by @komatits
>>>> <https://github.com/komatits>. Could it be that one of the default
>>>> values you changed affect this example?
>>>>
>>>> —
>>>> You are receiving this because you were mentioned.
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/geodynamics/specfem2d/pull/727#issuecomment-297073097>,
>>>>
>>>>
>>>>
>>>> or mute the thread
>>>> <https://github.com/notifications/unsubscribe-auth/AFjDKZ-kTEQYf77igEO5qVv7a7FnWUx1ks5rzhS-gaJpZM4NGITG>.
>>>>
>>>>
>>>>
>>>>
>>>
>>
>

-- 
Dimitri Komatitsch, CNRS Research Director (DR CNRS)
Laboratory of Mechanics and Acoustics, Marseille, France
http://komatitsch.free.fr
